reana/reana_dev/utils.py

```diff
@@ -335,3 +337,93 @@ def get_prefixed_component_name(component):
     :return: Prefixed name.
     """
     return "-".join([INSTANCE_NAME, component])
+
+
+def fetch_latest_pypi_version(package):
+    """Fetch latest released version of a package."""
+    import requests
+
+    pypi_rc_info = requests.get("https://pypi.python.org/pypi/{}/json".format(package))
+    return sorted(pypi_rc_info.json()["releases"].keys())[-1]
+
+
+def update_module_in_cluster_components(
+    module,
+    new_version,
+    components_to_update=None,
+    use_latest_known_tag=True,
+    force_feature_branch=False,
+):
+    """Update the specified module version in all affected components."""
+
+    def _check_is_feature_branch(component):
+        branch = get_current_branch(get_srcdir(component))
+        if branch == "master":
+            click.secho(
+                f"[ERROR] Component {component} current branch is master.",
+                err=True,
+                fg="red",
+            ),
+            sys.exit(1)
+
+    def _replace_pypi_version(
+        module_grep=None, file_=None, find=None, replace=None, component=None
+    ):
+        cmd = (
+            f'LN=`cat {file_} | grep -n -e "{module_grep}" | cut -f1 -d: `'
+            f' && sed -i.bk "`echo $LN`s/{find}/{replace}/"'
+            f" {file_} && rm {file_}.bk"
+        )
+        run_command(cmd, component)
+
+    updatable_components = {
+        "reana-commons": COMPONENTS_USING_SHARED_MODULE_COMMONS + ["reana-client"],
+        "reana-db": COMPONENTS_USING_SHARED_MODULE_DB,
+    }[module]
+
+    if components_to_update:
+        components_to_update = set(components_to_update).intersection(
+            set(updatable_components)
+        )
+    else:
+        components_to_update = updatable_components
+
+    if components_to_update and not use_latest_known_tag:
+        tag = run_command("git tag --points-at", module, return_output=True)
+        if not tag:
+            click.secho(
+                f"[ERROR] Last commit of {module} does not point to a tag."
+                " Use `--use-latest-known-tag` if you want to proceed using the lastest"
+                " known tag.",
+                err=True,
+                fg="red",
+            ),
+            sys.exit(1)
```

Comment:
Could you extract this logic to a function? For example `is_last_commit_tagged(component)` and return a boolean, so we customise the error message in each case? We will need this in many other places.